sql/binlog.cc

```diff
@@ -731,9 +731,17 @@ class binlog_cache_data
   {
     if (!is_binlog_empty())
     {
+#ifdef WITH_WSREP
```

Comment:
Wouldn't it be easier to check here
```WSREP_EMULATE_BINLOG```
and update variables or not?
Then we would not need all the stuff from ```THD::binlog_setup_trx_data()```

Reply:
Yes. It is easier. But what stopped me from doing it is that `THD` is not available in this part of the code, thereby making us depend on the `current_thd` to get the thread pointer. As `current_thd` has had problems with performance in the past, I though it is better to not register the status vars in `binlog_setup_trx_data()` is the best solution at hand.

Reply:
Fixed! Used `wsrep_emulate_bin_log` instead of `WSREP_EMULATE_BINLOG`.